lua/neorg/modules/core/norg/qol/toc/module.lua

```diff
@@ -0,0 +1,376 @@
+
+- `lua neorg.modules.get_module('core.norg.qol.toc').display_toc()`
+
+Calling display_toc with true (`...display_toc(true)`) will generate a split view of the Toc.
+--]]
+require("neorg.modules.base")
+
+local module = neorg.modules.create("core.norg.qol.toc")
+
+module.setup = function()
+    return {
+        success = true,
+        requires = {
+            "core.integrations.treesitter",
+            "core.ui",
+            "core.keybinds",
+            "core.mode",
+            "core.norg.esupports.hop",
+            "core.autocommands",
+        },
+    }
+end
+
+module.load = function()
+    module.required["core.keybinds"].register_keybinds(module.name, { "hop-toc-link", "close" })
+
+    module.required["core.autocommands"].enable_autocommand("BufLeave")
+    module.required["core.autocommands"].enable_autocommand("BufEnter")
+    module.required["core.autocommands"].enable_autocommand("BufDelete")
+    module.required["core.autocommands"].enable_autocommand("QuitPre")
+end
+
+module.config.public = {
+    -- If you use `default_toc_mode = "split"`, you can decide if you want the toc to persist until you close it.
+    -- Else, it'll close automatically when following a link
+    close_split_on_jump = false,
+    -- Where to place the TOC (`"left"`/`"right"`)
+    toc_split_placement = "left",
+}
+
+module.private = {
+    toc_bufnr = nil,
+    cur_bnr = nil,
+    cur_win = nil,
+
+    close_buffer = function(opts)
+        opts = opts or {}
+
+        if not opts.keep_buf then
+            vim.api.nvim_buf_delete(0, { force = true })
+            module.private.cur_bnr = nil
+            module.private.toc_bufnr = nil
+        end
+    end,
+}
+
+module.public = {
+    follow_link_toc = function()
+        -- Get the link node from cursor
+        local node = module.required["core.norg.esupports.hop"].lookahead_link_node()
+
+        if not node then
+            return
+        end
+
+        -- Parse the link before jumping
+        local parsed_link = module.required["core.norg.esupports.hop"].parse_link(node, module.private.toc_bufnr)
+
+        -- Follow the link on main norg file
+        vim.api.nvim_set_current_win(module.private.cur_win)
+        -- vim.api.nvim_win_set_buf(0, module.private.cur_bnr)
+        module.required["core.norg.esupports.hop"].follow_link(node, nil, parsed_link)
+    end,
+
+    --- Find a Table of Contents insertions in the document and returns its data
+    --- @return table A table that consist of two values: { item, parameters }.
+    --- Parameters can be nil if no parameters to the insertion were given.
+    find_toc = function()
+        -- Extract any insertion that has a ToC value in it
+        local query = vim.treesitter.parse_query(
+            "norg",
+            [[
+            (insertion
+            (insertion_prefix)
+            item: (capitalized_word) @item
+            parameters: (paragraph_segment)? @parameters
+            (#match? @item "^[tT][oO][cC]$")
```

Comment:
```suggestion
            (#match? @item "^T[oO][cC]$")
```
Needs to be capitalized according to the current NFF spec (insertions are capitalized, variables are lowercase)